allennlp/training/tensorboard_writer.py

```diff
@@ -1,23 +1,21 @@
-from typing import Any, Callable, Dict, List, Optional, Set
+from typing import Dict, Optional, Callable
+from numbers import Number
 import logging
 import os
 
 from tensorboardX import SummaryWriter
 import torch
 
-from allennlp.common.from_params import FromParams
-from allennlp.data import TensorDict
-from allennlp.nn import util as nn_util
-from allennlp.training.optimizers import Optimizer
-from allennlp.training import util as training_util
 from allennlp.models.model import Model
 
+from allennlp.training.log_writer import LogWriter
+
 logger = logging.getLogger(__name__)
 
 
-class TensorBoardWriter(FromParams):
+class TensorBoardWriter(LogWriter):
     """
-    Class that handles TensorBoard (and other) logging.
+    Class that handles TensorBoard logging.
```

Comment:
The API docs here are not building correctly, and I think it's because there's a blank line on line 24 (not your fault).

It's actually okay to have a blank line there, but we need to have a blank line in between every different parameter description. In other words, we need add a blank line after lines 26, 28, 43, etc...